qcodes/dataset/sqlite_base.py

```diff
@@ -112,6 +116,56 @@ class ConnectionPlus(wrapt.ObjectProxy):
 SomeConnection = Union[sqlite3.Connection, ConnectionPlus]
 
 
+def upgrader(func: Callable[[SomeConnection], None]):
+    """
+    Decorator for database version upgrade functions. An upgrade function
+    must have the name `perform_db_upgrade_N_to_M` where N = M-1.
+    The upgrade function must either perform the upgrade and return (no return
+    values allowed) or fail to perform the upgrade, in which case it must raise
+    a RuntimeError. A failed upgrade must be completely rolled back before the
+    RuntimeError is raises.
+
+    The decorator takes care of logging about the upgrade and managing the
+    database versioning.
+    """
+    name_comps = func.__name__.split('_')
+    if not len(name_comps) == 6:
+        raise NameError('Decorated function not a valid upgrader. '
+                        'Must have name "perform_db_upgrade_N_to_M"')
+    if not ''.join(name_comps[:3]+[name_comps[4]]) == 'performdbupgradeto':
+        raise NameError('Decorated function not a valid upgrader. '
+                        'Must have name "perform_db_upgrade_N_to_M"')
+    from_version = int(name_comps[3])
+    to_version = int(name_comps[5])
+
+    if not to_version == from_version+1:
+        raise ValueError(f'Invalid upgrade versions in function name: '
+                         f'{func.__name__}. Can only upgrade from version N'
+                         'to version N+1')
```

Comment:
does it add value to append something like `this function tried to upgrade from {from_version} to {to_version}`

Reply:
Sure.